ocp-demo-frontend code sample doesn't compile on JDK9

Building code samples on JDK9 results in the following: 
```
[INFO] Compiling 9 source files to /home/jenkins/hazelcast-code-samples/hazelcast-integration/openshift/client-apps/ocp-demo-frontend/target/classes
java.lang.NoSuchFieldError: members_field
	at lombok.javac.handlers.JavacHandlerUtil.fixMethodMirror(JavacHandlerUtil.java:962)
	at lombok.javac.handlers.JavacHandlerUtil.injectMethod(JavacHandlerUtil.java:953)
	at lombok.javac.handlers.HandleConstructor.generateConstructor(HandleConstructor.java:242)
	at lombok.javac.handlers.HandleBuilder.handle(HandleBuilder.java:158)
	at lombok.javac.HandlerLibrary$AnnotationHandlerContainer.handle(HandlerLibrary.java:109)
	at lombok.javac.HandlerLibrary.handleAnnotation(HandlerLibrary.java:244)
	at lombok.javac.JavacTransformer$AnnotationVisitor.visitAnnotationOnType(JavacTransformer.java:87)
	at lombok.javac.JavacNode.traverse(JavacNode.java:120)
	at lombok.javac.JavacAST.traverseChildren(JavacAST.java:143)
	at lombok.javac.JavacNode.traverse(JavacNode.java:83)
	at lombok.javac.JavacAST.traverseChildren(JavacAST.java:143)
	at lombok.javac.JavacNode.traverse(JavacNode.java:78)
	at lombok.javac.JavacAST.traverse(JavacAST.java:139)
	at lombok.javac.JavacTransformer.transform(JavacTransformer.java:71)
	at lombok.javac.apt.LombokProcessor.process(LombokProcessor.java:294)
	at lombok.core.AnnotationProcessor$JavacDescriptor.process(AnnotationProcessor.java:114)
	at lombok.core.AnnotationProcessor.process(AnnotationProcessor.java:164)
	at lombok.launch.AnnotationProcessorHider$AnnotationProcessor.process(AnnotationProcessor.java:74)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:968)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:884)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.access$2200(JavacProcessingEnvironment.java:108)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1206)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1315)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1246)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:922)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:100)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:142)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:96)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:90)
	at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:125)
	at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:171)
	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:886)
	at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:129)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Sample Code ........................................ SUCCESS [ 15.896 s]
[INFO] Clients ............................................ SUCCESS [  4.553 s]
[INFO] Clients - Basic .................................... SUCCESS [ 11.109 s]
[INFO] Hazelcast EE code sample helper classes ............ SUCCESS [  0.413 s]
[INFO] DistributedMap ..................................... SUCCESS [  0.010 s]
[INFO] DistributedMap - Near Cache ........................ SUCCESS [  0.321 s]
[INFO] Client Near Cache .................................. SUCCESS [  0.275 s]
[INFO] Clients - Rest Client .............................. SUCCESS [ 26.804 s]
[INFO] client-statistics .................................. SUCCESS [  0.174 s]
[INFO] user-code-deployment ............................... SUCCESS [  0.023 s]
[INFO] user-code-deployment-member ........................ SUCCESS [  0.133 s]
[INFO] user-code-deployment-client ........................ SUCCESS [  0.159 s]
[INFO] Cluster ............................................ SUCCESS [  4.113 s]
[INFO] Cluster Quorum ..................................... SUCCESS [ 11.246 s]
[INFO] Cluster Quorum sample for queue and lock with XML configuration SUCCESS [  0.279 s]
[INFO] Cluster State ...................................... SUCCESS [  0.194 s]
[INFO] Demo ............................................... SUCCESS [  0.554 s]
[INFO] Distributed Collections ............................ SUCCESS [  0.012 s]
[INFO] Distributed Collections - BlockingQueue ............ SUCCESS [  0.132 s]
[INFO] Distributed Collections - Bounded BlockingQueue .... SUCCESS [  0.157 s]
[INFO] Distributed Collections - Item Listeners ........... SUCCESS [  0.255 s]
[INFO] Distributed Collections - List ..................... SUCCESS [  0.265 s]
[INFO] Distributed Collections - QueueStore ............... SUCCESS [  0.287 s]
[INFO] Distributed Collections - RingbufferStore .......... SUCCESS [  0.240 s]
[INFO] Distributed Collections - Ringbuffer ............... SUCCESS [  0.288 s]
[INFO] Distributed Collections - Set ...................... SUCCESS [  0.215 s]
[INFO] Distributed Executor ............................... SUCCESS [  0.049 s]
[INFO] Distributed Executor - Basic Executor .............. SUCCESS [  0.218 s]
[INFO] Distributed Executor - Durable Executor Service .... SUCCESS [  0.218 s]
[INFO] Distributed Executor - Scheduling on key owner durable SUCCESS [  0.162 s]
[INFO] Distributed Executor - Scheduling on specific member SUCCESS [  0.137 s]
[INFO] Distributed Executor - Scheduling named task ....... SUCCESS [  0.168 s]
[INFO] Distributed Executor - Scheduling stateful task .... SUCCESS [  0.395 s]
[INFO] Distributed Executor - Scheduled Executor retrieve lost future SUCCESS [  0.264 s]
[INFO] Distributed Executor - Scheduled Executor Service retrieve all tasks SUCCESS [  0.143 s]
[INFO] Distributed Executor - Executing on all members .... SUCCESS [  0.161 s]
[INFO] Distributed Executor - Executing on key owner ...... SUCCESS [  0.135 s]
[INFO] Distributed Executor - Executing on lite members ... SUCCESS [  0.132 s]
[INFO] Distributed Executor - Executing on specific member  SUCCESS [  0.127 s]
[INFO] Distributed Executor - Execution Callback .......... SUCCESS [  0.121 s]
[INFO] Distributed Executor - Futures ..................... SUCCESS [  0.125 s]
[INFO] Distributed Executor - Scale Out ................... SUCCESS [  0.145 s]
[INFO] Distributed Executor - Scale Up .................... SUCCESS [  0.138 s]
[INFO] Aggregations ....................................... SUCCESS [  0.163 s]
[INFO] DistributedMap - Backup ............................ SUCCESS [  0.154 s]
[INFO] DistributedMap - Basics ............................ SUCCESS [  0.133 s]
[INFO] DistributedMap - Continuous Query .................. SUCCESS [  0.139 s]
[INFO] Continuous Query Cache (CQC) ....................... SUCCESS [  1.393 s]
[INFO] DistributedMap - Criteria API ...................... SUCCESS [  0.184 s]
[INFO] DistributedMap - Custom Attributes ................. SUCCESS [  0.290 s]
[INFO] DistributedMap - Data Locality ..................... SUCCESS [  0.204 s]
[INFO] DistributedMap - EntryListener ..................... SUCCESS [  0.168 s]
[INFO] DistributedMap - EntryProcessor .................... SUCCESS [  0.224 s]
[INFO] DistributedMap - Eviction .......................... SUCCESS [  0.201 s]
[INFO] DistributedMap - Hashcode and equals ............... SUCCESS [  0.131 s]
[INFO] DistributedMap - In Memory Format .................. SUCCESS [  0.170 s]
[INFO] DistributedMap - Index ............................. SUCCESS [  0.239 s]
[INFO] DistributedMap - Locking ........................... SUCCESS [  0.179 s]
[INFO] DistributedMap - MapInterceptor .................... SUCCESS [  0.110 s]
[INFO] DistributedMap - MapStore .......................... SUCCESS [ 30.282 s]
[INFO] DistributedMap - Multimap .......................... SUCCESS [  0.109 s]
[INFO] DistributedMap - PagingPredicate ................... SUCCESS [  0.127 s]
[INFO] DistributedMap - MapPartitionLostListener .......... SUCCESS [  0.108 s]
[INFO] DistributedMap - Query Collections and Arrays ...... SUCCESS [  0.119 s]
[INFO] DistributedMap - SqlQuery .......................... SUCCESS [  0.120 s]
[INFO] Fast Aggregations .................................. SUCCESS [  0.141 s]
[INFO] Fast Aggregations .................................. SUCCESS [  0.148 s]
[INFO] Distributed Primitives ............................. SUCCESS [  0.025 s]
[INFO] Distributed Primitives - IAtomicLong ............... SUCCESS [  0.102 s]
[INFO] Distributed Primitives - IAtomicLong Functions ..... SUCCESS [  0.108 s]
[INFO] Distributed Primitives - IAtomicReference .......... SUCCESS [  0.087 s]
[INFO] Distributed Primitives - CardinalityEstimator ...... SUCCESS [  0.147 s]
[INFO] Distributed Primitives - Condition ................. SUCCESS [  0.095 s]
[INFO] Distributed Primitives - CountdownLatch ............ SUCCESS [  0.121 s]
[INFO] Distributed Primitives - IdGenerator ............... SUCCESS [  0.121 s]
[INFO] Distributed Primitives - FlakeIdGenerator .......... SUCCESS [  0.115 s]
[INFO] Distributed Primitives - PNCounter ................. SUCCESS [  0.139 s]
[INFO] Distributed Primitives - Lock ...................... SUCCESS [  0.118 s]
[INFO] Distributed Primitives - Semaphore ................. SUCCESS [  0.108 s]
[INFO] Distributed Topic .................................. SUCCESS [  0.020 s]
[INFO] Distributed Topic - Basic Pub Sub .................. SUCCESS [  0.145 s]
[INFO] Distributed Topic - Reliable Topic ................. SUCCESS [  0.101 s]
[INFO] Distributed Topic - Scaling ........................ SUCCESS [  0.116 s]
[INFO] Enterprise ......................................... SUCCESS [ 10.704 s]
[INFO] Client SSL ......................................... SUCCESS [  0.202 s]
[INFO] Enterprise - HD IMap ............................... SUCCESS [  0.231 s]
[INFO] Enterprise - HD Memory Client / Server ............. SUCCESS [  0.027 s]
[INFO] hd-memory-examples-server .......................... SUCCESS [  4.288 s]
[INFO] hd-memory-examples-client .......................... SUCCESS [  6.967 s]
[INFO] jcache ............................................. SUCCESS [  3.133 s]
[INFO] Enterprise - HiDensity Cache ....................... SUCCESS [  1.337 s]
[INFO] Hot Restart ........................................ SUCCESS [  0.280 s]
[INFO] Enterprise - Security Interceptor .................. SUCCESS [  0.165 s]
[INFO] Enterprise - Socket Interceptor .................... SUCCESS [  0.237 s]
[INFO] Enterprise - Wan Replication ....................... SUCCESS [ 42.125 s]
[INFO] hazelcast-integration .............................. SUCCESS [  0.021 s]
[INFO] amazon-ec2-vagrant-chef ............................ SUCCESS [  0.147 s]
[INFO] amazon-elasticbeanstalk ............................ SUCCESS [01:12 min]
[INFO] Hazelcast Dynacache Integration Example ............ SUCCESS [02:03 min]
[INFO] Scaling Hazelcast IMDG cluster with Docker-Compose . SUCCESS [  9.028 s]
[INFO] Hazelcast Filter Based Session Replication ......... SUCCESS [  0.499 s]
[INFO] Hibernate 2nd Level Cache .......................... SUCCESS [ 29.105 s]
[INFO] Hibernate JPA 2nd Level Cache ...................... SUCCESS [ 26.560 s]
[INFO] Hazelcast JCA Example .............................. SUCCESS [ 12.100 s]
[INFO] Manager Based Session Replication Example .......... SUCCESS [  0.903 s]
[INFO] hazelcast-mongodb-integration-sample ............... SUCCESS [  3.998 s]
[INFO] openshift .......................................... SUCCESS [  0.023 s]
[INFO] client-apps ........................................ SUCCESS [  0.024 s]
[INFO] ocp-entry-processor ................................ SUCCESS [  0.140 s]
[INFO] ocp-demo-frontend .................................. FAILURE [01:30 min]
[INFO] pcf-integration .................................... SKIPPED
[INFO] Spark Connector Example ............................ SKIPPED
[INFO] Spring Cache Manager Example ....................... SKIPPED
[INFO] Hazelcast Spring Configuration Example ............. SKIPPED
[INFO] Spring Data Hazelcast Chemistry Parent (pom) ....... SKIPPED
[INFO] Spring Data Hazelcast Chemistry common (jar) ....... SKIPPED
[INFO] Spring Data Hazelcast Chemistry server (jar) ....... SKIPPED
[INFO] Spring Data Hazelcast Chemistry client (jar) ....... SKIPPED
[INFO] Spring Data Integration ............................ SKIPPED
[INFO] Spring Data JPA to Hazelcast ....................... SKIPPED
[INFO] Spring Data JPA to Hazelcast (Shared utilities) .... SKIPPED
[INFO] Spring Data JPA to Hazelcast (RDBMS database) ...... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("Before" domain) ..... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("Before" JPA repo) ... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("Before" JPA service)  SKIPPED
[INFO] Spring Data JPA to Hazelcast ("Before" main) ....... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("Before") ............ SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After" domain) ...... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After" JPA repo) .... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After" JPA service) . SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After" KV repo) ..... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After" KV service) .. SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After" HZ main) ..... SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After" main) ........ SKIPPED
[INFO] Spring Data JPA to Hazelcast ("After") ............. SKIPPED
[INFO] Spring Hibernate 2nd Level Cache Example ........... SKIPPED
[INFO] spring-jcache ...................................... SKIPPED
[INFO] hazelcast-spring-security-sample ................... SKIPPED
[INFO] Spring Transaction Manager Example ................. SKIPPED
[INFO] springaware-annotation ............................. SKIPPED
[INFO] Spring Boot: Hazelcast via Spring Caching Abstraction SKIPPED
[INFO] Spring Boot: Hazelcast via Spring JCache ........... SKIPPED
[INFO] Partition Groups with Eureka ....................... SKIPPED
[INFO] Partition Groups with Eureka (Common) .............. SKIPPED
[INFO] Partition Groups with Eureka (Eureka Server) ....... SKIPPED
[INFO] Partition Groups with Eureka (Eureka Client) ....... SKIPPED
[INFO] Partition Groups with Eureka (Hazelcast Server) .... SKIPPED
[INFO] Partition Groups with Eureka (Hazelcast Client) .... SKIPPED
[INFO] Spring Boot: Hazelcast for Session Replication ..... SKIPPED
[INFO] Spring Boot: HTTP Session Analysis ................. SKIPPED
[INFO] Spring Boot: HTTP Session Analysis: Common ......... SKIPPED
[INFO] Spring Boot: HTTP Session Analysis: Client ......... SKIPPED
[INFO] Spring Boot: HTTP Session Analysis: Server ......... SKIPPED
[INFO] Tricolor ........................................... SKIPPED
[INFO] JCache 1.1 ......................................... SKIPPED
[INFO] JCache:1.1: Times Table ............................ SKIPPED
[INFO] Times Table: Common ................................ SKIPPED
[INFO] Times Table: Server ................................ SKIPPED
[INFO] Times Table: JCache with Spring .................... SKIPPED
[INFO] Times Table: JCache standard style, no Spring ...... SKIPPED
[INFO] jmx ................................................ SKIPPED
[INFO] Learning Basics .................................... SKIPPED
[INFO] Learning Basics - Logging Configuration ............ SKIPPED
[INFO] Learning Basics - Configure Programmatic ........... SKIPPED
[INFO] Learning Basics - Configure XML .................... SKIPPED
[INFO] Learning Basics - Destroying instances ............. SKIPPED
[INFO] Learning Basics - Loading instances ................ SKIPPED
[INFO] Learning Basics - Multiple Hazelcast Instances ..... SKIPPED
[INFO] Learning Basics - Unique Names ..................... SKIPPED
[INFO] Learning Basics - Wildcard Configuration ........... SKIPPED
[INFO] Learning Basics - Configuration of Data Structures at Runtime SKIPPED
[INFO] Map/Reduce ......................................... SKIPPED
[INFO] Monitoring ......................................... SKIPPED
[INFO] Monitoring - Cluster/Member Safety Examples ........ SKIPPED
[INFO] Near Cache ......................................... SKIPPED
[INFO] Near Cache: Fraud Detection ........................ SKIPPED
[INFO] Near Cache: Fraud Detection Common ................. SKIPPED
[INFO] Near Cache: Fraud Detection Server ................. SKIPPED
[INFO] Nea -Cache: Fraud Detection Client abstract base ... SKIPPED
[INFO] Near Cache: Fraud Detection Client with near-cache . SKIPPED
[INFO] Near Cache: Fraud Detection Client no near-cache ... SKIPPED
[INFO] Network Configuration .............................. SKIPPED
[INFO] Network Configuration - AWS ........................ SKIPPED
[INFO] Network Configuration - Firewall ................... SKIPPED
[INFO] Network Configuration - Groups ..................... SKIPPED
[INFO] Network Configuration - Group1 ..................... SKIPPED
[INFO] Network Configuration - Group2 ..................... SKIPPED
[INFO] Network Configuration - jclouds .................... SKIPPED
[INFO] Network Configuration - jclouds partition group .... SKIPPED
[INFO] Network Configuration - Multicast .................. SKIPPED
[INFO] Network Configuration - Multicast Plugin ........... SKIPPED
[INFO] Network Configuration - Partition Group ............ SKIPPED
[INFO] Network Configuration - Port ....................... SKIPPED
[INFO] Network Configuration - SSL ........................ SKIPPED
[INFO] Network Configuration - Symmetric Encryption ....... SKIPPED
[INFO] Network Configuration - TcpIp ...................... SKIPPED
[INFO] org-website-samples ................................ SKIPPED
[INFO] osgi ............................................... SKIPPED
[INFO] Replicated-Map ..................................... SKIPPED
[INFO] ReplicatedMap - Basics ............................. SKIPPED
[INFO] ReplicatedMap - EntryListener ...................... SKIPPED
[INFO] ReplicatedMap - In Memory Format ................... SKIPPED
[INFO] Querying ........................................... SKIPPED
[INFO] Querying: Projection on Keys ....................... SKIPPED
[INFO] Serialization ...................................... SKIPPED
[INFO] Serialization: ByteArray Serializer ................ SKIPPED
[INFO] Serialization: DataSerializable .................... SKIPPED
[INFO] Serialization: Externalizable ...................... SKIPPED
[INFO] Serialization: Global Serializer ................... SKIPPED
[INFO] Serialization: Hazelcast Airlines .................. SKIPPED
[INFO] Serialization: Hazelcast Airlines : The Code ....... SKIPPED
[INFO] Serialization: Hazelcast Airlines : Grid Node ...... SKIPPED
[INFO] Serialization: HazelcastInstanceAware .............. SKIPPED
[INFO] Serialization: IdentifiedDataSerializable .......... SKIPPED
[INFO] Serialization: Kryo Serializer ..................... SKIPPED
[INFO] Serialization: ManagedContext ...................... SKIPPED
[INFO] Serialization: Portable ............................ SKIPPED
[INFO] Serialization: Protobuf Serializer ................. SKIPPED
[INFO] Serialization: Serializable ........................ SKIPPED
[INFO] Serialization: Stream Serializer ................... SKIPPED
[INFO] Serialization: UserContext ......................... SKIPPED
[INFO] SPI ................................................ SKIPPED
[INFO] SPI: Backups ....................................... SKIPPED
[INFO] SPI: CounterService.Container ...................... SKIPPED
[INFO] Discovery SPI ...................................... SKIPPED
[INFO] SPI: Getting started ............................... SKIPPED
[INFO] SPI: Notification .................................. SKIPPED
[INFO] SPI: Partition Migration ........................... SKIPPED
[INFO] SPI: Priority Queue ................................ SKIPPED
[INFO] Priority Queue Core ................................ SKIPPED
[INFO] Priority Queue Server .............................. SKIPPED
[INFO] SPI: Proxy ......................................... SKIPPED
[INFO] SPI: Split-brain healing ........................... SKIPPED
[INFO] Transactions ....................................... SKIPPED
[INFO] Transactions - Transaction Basics .................. SKIPPED
[INFO] Transactions - Transactional Task .................. SKIPPED
[INFO] Transactions - XA Transaction ...................... SKIPPED
[INFO] Variable replacers ................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10:26 min
[INFO] Finished at: 2018-08-14T08:09:14+00:00
[INFO] Final Memory: 114M/381M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project ocp-demo-frontend: Fatal error compiling: java.lang.NoClassDefFoundError: com/sun/tools/javac/file/BaseFileObject: com.sun.tools.javac.file.BaseFileObject -> [Help 1]
```

Lombok library has probably problems on JDK9. IMO, Lombok should not be present in the code samples in a first place. Code samples should be as simple as possible. Lombok is quite a magic that not all developers (including me) are familiar with. I would strongly vote for removing it.

--- a/hazelcast-integration/amazon-ec2-vagrant-chef/pom.xml
+++ b/hazelcast-integration/amazon-ec2-vagrant-chef/pom.xml
@@ -23,6 +23,7 @@
         <plugins>
             <plugin>
                 <artifactId>maven-assembly-plugin</artifactId>
+                <version>${maven.assembly.plugin.version}</version>
                 <configuration>
                     <archive>
                         <manifest>

--- a/hazelcast-integration/docker-compose/pom.xml
+++ b/hazelcast-integration/docker-compose/pom.xml
@@ -29,6 +29,7 @@
             <plugin>
                 <!-- builds fat jar to be deployed as docker image -->
                 <artifactId>maven-assembly-plugin</artifactId>
+                <version>${maven.assembly.plugin.version}</version>
                 <configuration>
                     <archive>
                         <manifest>

--- a/hazelcast-integration/jca-ra/pom.xml
+++ b/hazelcast-integration/jca-ra/pom.xml
@@ -79,6 +79,10 @@
             <version>2.0.0</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.spec.javax.annotation</groupId>
+            <artifactId>jboss-annotations-api_1.1_spec</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

--- a/hazelcast-integration/openshift/client-apps/ocp-demo-frontend/pom.xml
+++ b/hazelcast-integration/openshift/client-apps/ocp-demo-frontend/pom.xml
@@ -74,7 +74,6 @@
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
-            <version>${lombok.version}</version>
             <optional>true</optional>
         </dependency>
 

--- a/hazelcast-integration/openshift/client-apps/pom.xml
+++ b/hazelcast-integration/openshift/client-apps/pom.xml
@@ -25,7 +25,6 @@
 
         <spring.version>4.3.7.RELEASE</spring.version>
         <spring.boot.version>1.5.2.RELEASE</spring.boot.version>
-        <lombok.version>1.16.14</lombok.version>
 
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>

--- a/hazelcast-integration/spring-configuration/pom.xml
+++ b/hazelcast-integration/spring-configuration/pom.xml
@@ -37,5 +37,10 @@
             <artifactId>spring-core</artifactId>
             <version>${spring.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.spec.javax.annotation</groupId>
+            <artifactId>jboss-annotations-api_1.2_spec</artifactId>
+            <version>1.0.2.Final</version>
+        </dependency>
     </dependencies>
 </project>

--- a/hazelcast-integration/spring-data-hazelcast-chemistry-sample/pom.xml
+++ b/hazelcast-integration/spring-data-hazelcast-chemistry-sample/pom.xml
@@ -42,6 +42,11 @@
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.spec.javax.annotation</groupId>
+            <artifactId>jboss-annotations-api_1.2_spec</artifactId>
+            <version>1.0.2.Final</version>
+        </dependency>
     </dependencies>
 
     <dependencyManagement>

--- a/hazelcast-integration/spring-data-jpa-hazelcast-migration/pom.xml
+++ b/hazelcast-integration/spring-data-jpa-hazelcast-migration/pom.xml
@@ -73,6 +73,11 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
+            <artifactId>jaxb-api</artifactId>
+            <version>2.3.0</version>
+        </dependency>
     </dependencies>
 
     <dependencyManagement>

--- a/jcache-1.1/times-table/jcache-standard/pom.xml
+++ b/jcache-1.1/times-table/jcache-standard/pom.xml
@@ -40,7 +40,7 @@
         <plugins>
             <plugin>
                 <artifactId>maven-assembly-plugin</artifactId>
-                <version>2.6</version>
+                <version>${maven.assembly.plugin.version}</version>
                 <configuration>
                     <appendAssemblyId>false</appendAssemblyId>
                     <archive>

--- a/network-configuration/jclouds-partitiongroup/pom.xml
+++ b/network-configuration/jclouds-partitiongroup/pom.xml
@@ -57,6 +57,7 @@
         <plugins>
             <plugin>
                 <artifactId>maven-assembly-plugin</artifactId>
+                <version>${maven.assembly.plugin.version}</version>
                 <executions>
                     <execution>
                         <phase>package</phase>

--- a/network-configuration/jclouds/pom.xml
+++ b/network-configuration/jclouds/pom.xml
@@ -52,6 +52,7 @@
         <plugins>
             <plugin>
                 <artifactId>maven-assembly-plugin</artifactId>
+                <version>${maven.assembly.plugin.version}</version>
                 <executions>
                     <execution>
                         <phase>package</phase>

--- a/pom.xml
+++ b/pom.xml
@@ -25,6 +25,9 @@
 
         <maven.compiler.plugin.version>3.5.1</maven.compiler.plugin.version>
         <maven.checkstyle.plugin.version>2.15</maven.checkstyle.plugin.version>
+        <maven.assembly.plugin.version>3.0.0</maven.assembly.plugin.version>
+        
+        <lombok.version>1.16.22</lombok.version>
     </properties>
 
     <repositories>
@@ -119,6 +122,16 @@
             </plugins>
         </pluginManagement>
     </build>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.projectlombok</groupId>
+                <artifactId>lombok</artifactId>
+                <version>${lombok.version}</version>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
 
     <profiles>
         <profile>

--- a/serialization/hazelcast-airlines/the-code/src/main/java/com/hazelcast/samples/serialization/hazelcast/airlines/MyKey.java
+++ b/serialization/hazelcast-airlines/the-code/src/main/java/com/hazelcast/samples/serialization/hazelcast/airlines/MyKey.java
@@ -5,7 +5,6 @@ import java.time.LocalDate;
 
 import lombok.AllArgsConstructor;
 import lombok.Data;
-import lombok.NoArgsConstructor;
 
 /**
  * <p>The key for a flight object. Keys here are
@@ -14,12 +13,14 @@ import lombok.NoArgsConstructor;
  */
 @AllArgsConstructor
 @Data
-@NoArgsConstructor
 @SuppressWarnings("serial")
 public class MyKey implements Comparable<MyKey>, Serializable {
 
     private String code;
     private LocalDate date;
+
+    public MyKey() {
+    }
 
     /**
      * <p>Sort ascending, on code then date

--- a/serialization/hazelcast-airlines/the-code/src/main/java/com/hazelcast/samples/serialization/hazelcast/airlines/Person.java
+++ b/serialization/hazelcast-airlines/the-code/src/main/java/com/hazelcast/samples/serialization/hazelcast/airlines/Person.java
@@ -4,7 +4,6 @@ import java.io.Serializable;
 
 import lombok.AllArgsConstructor;
 import lombok.Data;
-import lombok.NoArgsConstructor;
 
 /**
  * <p>Someone who is on a flight. Note that this object is
@@ -14,9 +13,12 @@ import lombok.NoArgsConstructor;
  */
 @AllArgsConstructor
 @Data
-@NoArgsConstructor
 @SuppressWarnings("serial")
 public class Person implements Serializable {
 
     private String name;
+
+    public Person() {
+    }
+
 }

